### ISSUE TITLE
How to treat null as a valid value?

### ISSUE THREAD

**Issue by honzajavorek** [crcn/sift.js#114]
I have a field, which can contain string or null as valid values. I want to match documents, which do not have the field at all. How can I do it with sift? I tried many combinations and nothing worked for me. `{$exists: false}` unfortunately seems to match `null` as well, so it's useless:

```js
{'field': {'$exists': false}}
```

Neither

```js
{'$and': [
  {'field': {'$not': {'$type': String}}},
  {'field': {'$ne': null}},
]}
```

works. I had to do this:

```js
{'field': function (value) {
  return value !== null and typeof value !== 'string'
}}
```

Seems pretty inconsistent with how Mongo treats `null`.

**Comment by crcn** [comment]
You're right, this is a design bug. Sift's version of `$exists` currently checks for the existence of anything except `null`, or `undefined`, which is inconsistent with Mongodb's implementation. 

I'll get around to this as soon as I can. This will likely be a `major` bump.

**Comment by crcn** [comment]
This behavior is now fixed in `5.0.0`. 🎉
